ceremonies/meetup-validation/src/meetup_scenario_tests.rs

```diff
@@ -0,0 +1,268 @@
+use super::*;
+use rstest::*;
+#[rstest(
+	meetup_size,
+	num_attendees,
+	n,
+	case(16, 16, 1),
+	case(16, 16, 5),
+	case(16, 16, 9),
+	case(16, 9, 0),
+	case(16, 9, 1),
+	case(16, 9, 5),
+	case(16, 3, 0),
+	case(16, 3, 1),
+	case(12, 7, 0),
+	case(12, 7, 3),
+	case(12, 5, 1),
+	case(12, 5, 2),
+	case(8, 8, 0),
+	case(8, 8, 1),
+	case(8, 8, 3),
+	case(8, 8, 5),
+	case(8, 6, 0),
+	case(8, 7, 1),
+	case(8, 5, 3),
+	case(3, 3, 0),
+	case(3, 3, 1),
+	case(3, 3, 2)
+)]
+fn n_attendees_do_not_attest_anyone(meetup_size: usize, num_attendees: usize, n: usize) {
+	let participants: Participants = (0..meetup_size).collect();
+	let mut participant_votes = vec![num_attendees as u32; num_attendees];
+	participant_votes.append(&mut vec![0u32; meetup_size - num_attendees]);
+	let mut participant_attestations: Attestations = vec![];
+	for i in 0..(num_attendees - n) {
+		participant_attestations.push((0..num_attendees).filter(|&j| j != i).collect());
+	}
+	for _ in 0..(meetup_size - num_attendees + n) {
+		participant_attestations.push(vec![]);
+	}
+	assert_eq!(
+		get_participant_judgements(
+			&participants,
+			&participant_votes,
+			&participant_attestations,
+			|n| n - 1,
+		)
+		.unwrap()
+		.legit,
+		(0..(num_attendees - n)).collect::<Vec<usize>>()
+	);
+}
+
+#[rstest(
+	meetup_size,
+	num_attendees,
+	n,
+	case(16, 16, 1),
+	case(16, 16, 5),
+	case(16, 16, 9),
+	case(16, 9, 0),
+	case(16, 9, 1),
+	case(16, 9, 5),
+	case(16, 3, 0),
+	case(16, 3, 1),
+	case(12, 7, 0),
+	case(12, 7, 3),
+	case(12, 5, 1),
+	case(12, 5, 2),
+	case(8, 8, 0),
+	case(8, 8, 1),
+	case(8, 8, 3),
+	case(8, 8, 5),
+	case(8, 6, 0),
+	case(8, 7, 1),
+	case(8, 5, 3),
+	case(3, 3, 0),
+	case(3, 3, 1),
+	case(3, 3, 2)
+)]
+fn attendee_is_not_attested_by_n_others(meetup_size: usize, num_attendees: usize, n: usize) {
+	let participants: Participants = (0..meetup_size).collect();
+	let mut participant_votes = vec![num_attendees as u32; num_attendees];
+	participant_votes.append(&mut vec![0u32; meetup_size - num_attendees]);
+	let mut participant_attestations: Attestations = vec![];
+
+	for i in 0..num_attendees {
+		let first_index = if i <= n { 1 } else { 0 };
```

Comment:
I can't understand the significance of the first index here.

Reply:
participant with index 0 is the guy that is not attested by n others. so everyone with index <= n will not attest the participant 0 and will therefore attest everyone starting at index 1